### DISCUSSION TITLE
STM32F103C8T6 (Bluepill) Upload Code Failed: Error "Access is denied", I have use the ST-LINK V2 to boot the Bluepill and try to upload my code but i could not upload it, but my friend is able to use his laptop to upload code but not mine. How do i solve it?

### DISCUSSION THREAD

**Discussion by WeiKangLim24** [stm32duino/Arduino_Core_STM32#2177] [Q&A] [answered]
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/88315090/805b36a2-1771-4257-874c-531fdcc1581b)

**Comment by WeiKangLim24** [comment]
I am currently using Window 11 on my Asus Laptop

**Reply by fpistm** [reply, marked as answer]
This issue is not link to the core, so please avoid to open issue. That's why I convert it as a discussion.
Seems you have some issue with USB. Like the STLink or the DFU utils can't access your board.
Driver issue? Limited permissions for USB device? Security policies?
I would advise to try only with STM32CubeProgrammer while it is nofunctional, no need to try within Arduino.
Maybe try to install drivers: https://www.st.com/en/development-tools/stsw-link009.html

**Comment by nopnop2002** [comment]
Check if ST-LINK exists in Device Manager.

![st-link](https://github.com/stm32duino/Arduino_Core_STM32/assets/6020549/b8ce10d2-03ac-410f-a734-1040b01d1a49)
